packages/core/src/state-management/__tests__/anchor-resuming-service.test.ts

```diff
@@ -0,0 +1,163 @@
+import { jest } from '@jest/globals'
+import { createCeramic } from '../../__tests__/create-ceramic.js'
+import { InMemoryAnchorService } from '../../anchor/memory/in-memory-anchor-service.js'
+import { Observable } from 'rxjs'
+import {
+  AnchorServiceResponse,
+  AnchorStatus,
+  IpfsApi,
+  SyncOptions,
+  TestUtils,
+} from '@ceramicnetwork/common'
+import { TileDocument } from '@ceramicnetwork/stream-tile'
+import tmp from 'tmp-promise'
+import { createIPFS } from '@ceramicnetwork/ipfs-daemon'
+import { AnchorResumingService } from '../anchor-resuming-service.js'
+
+describe('resumeRunningStatesFromAnchorRequestStore(...) method', () => {
+  jest.setTimeout(10000)
+
+  let ipfs: IpfsApi
+  let stateStoreDirectoryName: string
+
+  beforeAll(async () => {
+    ipfs = await createIPFS()
+  })
+
+  afterAll(async () => {
+    await ipfs.stop()
+  })
+
+  beforeEach(async () => {
+    stateStoreDirectoryName = await tmp.tmpName()
+  })
+
+  afterEach(async () => {
+    jest.resetAllMocks()
+  })
+
+  test('Requests anchors for streams with stored anchor requests', async () => {
+    const ceramic = await createCeramic(ipfs, {
+      anchorOnRequest: false,
+      stateStoreDirectory: stateStoreDirectoryName,
+    })
+
+    const anchorService = ceramic.repository.stateManager.anchorService as InMemoryAnchorService
+    const realRequestAnchor = anchorService.requestAnchor
+    const mockedRequestAnchor = jest.fn()
+    mockedRequestAnchor.mockImplementation(() => {
+      return new Observable<AnchorServiceResponse>()
+    })
+    // @ts-ignore
+    anchorService.requestAnchor = mockedRequestAnchor
+
+    const stream = await TileDocument.create(ceramic, { x: 1 }, null, {
+      anchor: true,
+    })
+
+    // Verify that the anchor was not requested (because we used the mocked anchorRequest(...) which does nothing)
+    expect(mockedRequestAnchor).toHaveBeenCalledTimes(1)
+    expect(stream.state.anchorStatus).toEqual(AnchorStatus.NOT_REQUESTED)
+
+    // Verify that the anchor request was stored
+    const storedAnchorRequests = await ceramic.repository.anchorRequestStore.list()
+    expect(storedAnchorRequests.length).toEqual(1)
+    expect(storedAnchorRequests[0].key).toEqual(stream.id)
+
+    // Set requestAnchor back to the proper implementation and resume anchors
+    anchorService.requestAnchor = realRequestAnchor
+    const anchorResumingService = new AnchorResumingService(
+      ceramic.loggerProvider.getDiagnosticsLogger()
+    )
+    await anchorResumingService.resumeRunningStatesFromAnchorRequestStore(ceramic.repository)
+
+    // Give the anchor service some time to work, sync the stream and check that the anchor had been requested
+    await TestUtils.delay(3000) // TODO(CDB-2090): use less brittle approach to waiting for this condition
```

Comment:
I'd love to not have to keep updating that PR for CDB-2090 as new PRs go in - any chance I could get a review on that soon and get it merged in so you could make the change here?